pkg/network/node/ovscontroller.go

```diff
@@ -769,33 +769,50 @@ func (oc *ovsController) ensureTunMAC() error {
 func (oc *ovsController) SetNamespaceEgressNormal(vnid uint32) error {
 	otx := oc.ovs.NewTransaction()
 	otx.DeleteFlows("table=100, reg0=%d", vnid)
+	otx.DeleteGroup(vnid)
 	return otx.Commit()
 }
 
 func (oc *ovsController) SetNamespaceEgressDropped(vnid uint32) error {
 	otx := oc.ovs.NewTransaction()
+	otx.DeleteGroup(vnid)
 	otx.DeleteFlows("table=100, reg0=%d", vnid)
 	otx.AddFlow("table=100, priority=100, reg0=%d, actions=drop", vnid)
 	return otx.Commit()
 }
 
-func (oc *ovsController) SetNamespaceEgressViaEgressIP(vnid uint32, nodeIP, mark string) error {
+func (oc *ovsController) SetNamespaceEgressViaEgressIPs(vnid uint32, egressIPsMetaData []egressIPMetaData) error {
 	otx := oc.ovs.NewTransaction()
 	otx.DeleteFlows("table=100, reg0=%d", vnid)
-	if nodeIP == "" {
-		// Namespace wants egress IP, but no node hosts it, so drop
-		otx.AddFlow("table=100, priority=100, reg0=%d, actions=drop", vnid)
-	} else if nodeIP == oc.localIP {
-		if err := oc.ensureTunMAC(); err != nil {
-			return err
+	otx.DeleteGroup(vnid)
+
+	var buildBuckets []string
+	for _, egressIPMetaData := range egressIPsMetaData {
+		if egressIPMetaData.nodeIP == oc.localIP {
+			if err := oc.ensureTunMAC(); err != nil {
+				return err
+			}
+			buildBuckets = []string{fmt.Sprintf("actions=set_field:%s->eth_dst,set_field:%s->pkt_mark,output:tun0", oc.tunMAC, egressIPMetaData.packetMark)}
+			//if one of the egressIPs is the IP of the node always use that egress otherwise it might not actually egress and get sent between nodes
```

Comment:
```suggestion
			// if one of the egressIPs is the IP of the node always use that egress otherwise it might not actually egress and get sent between nodes
```

Reply:
done